src/main/java/org/springframework/data/solr/core/SolrTemplate.java

```diff
@@ -89,8 +90,9 @@
  * @author Joachim Uhrlass
  * @author Francisco Spaeth
  * @author Shiradwade Sateesh Krishna
+ * @author Venil Noronha
  */
-public class SolrTemplate implements SolrOperations, InitializingBean, ApplicationContextAware {
+public class SolrTemplate implements SolrOperations, MulticoreSolrOperations, InitializingBean, ApplicationContextAware {

```

Comment:
Is SolrTemplate really playing this role? Shouldn't this be implemented in MulticoreSolrTemplate?
AFAIK we can create the Template in a configuration. So it would be possible to exchange the behavior as needed.